### ISSUE TITLE
Blank table header row in Drupal displays as blank on FE

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#11025]
## Describe the defect
The Drupal table_field form UI states that the top row of a table can be left blank if the table doesn't have headers. 

However, when Editors leave the top row blank, the FE template still renders a header row, it's just blank
- The header cells still have accessibility attributes like `scope="col"`, which would confuse the screen reader experience
- It doesn't make sense to sighted users either
- Mobile display doesn't have headers over each piece of data as intended

Example: https://www.va.gov/houston-health-care/health-services/patient-advocates/
![image.png](https://images.zenhubusercontent.com/61a671f5fc46c2a311655f75/3812fa14-c3c7-4efd-bc8f-f9c738715452)

## To Reproduce
1. Create a blank table (or just look at CMS edit view for example page above)
2. See blank header row rendered on VAgov FE
3. Inspect elements to see accessibility attributes on blank header cells

## AC / Expected behavior
- [ ] Update CMS table functionality to require content in the top row 
- [ ] Update help text so that editors are not encouraged to leave the top row blank, rather they should be told that the top row will be the table header row

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by laflannery** [comment]
@wesrowe the best solution would definitely be changing the Drupal behavior and not allowing editors to leave the top row blank - therefore all tables would have headers in both the top row and first column.  

Regarding your second question, yes it would be an improvement if we corrected the conditional (removing the blank headers) however I have yet to come across a table where only first column headers would be appropriate. If we did correct the conditional as a first step - would we also be able to adjust the Drupal behavior? Or are we thinking about this as an either or scenario?

**Comment by wesrowe** [comment]
I'm going to ask Drupal engineers to review the ticket and say whether it's possible to require data in the top row. If so we should do it, as that will make new tables better going forward.

@swirtSJW @dsasser, can we modify the table_field form to require data in the top row? Currently the help text says it's optional, and the result is non-508 compliant and looks weird on the FE.

**Comment by dsasser** [comment]
> @swirtSJW @dsasser, is it possible modify the table_field form to require data in the top row? Currently the help text says it's optional, and the result is non-508 compliant and looks weird on the FE.

@wesrowe this is possible but presents some challenges. The tablefield module is written to accept a blank row to mean "the first row that actually has data is not a header", so this would require a bit of work to rearchitect that out. Also, it is complicated by the fact that we are also trying to move the header selections to separate checkboxes in #10675 at the same time. The tablefield module has been patched heavily on VA, and I have a concern that if we continue to patch it, there will be a point where patches fail to apply because they have overlapping areas of concern. So we would only go this route if we were ok with navigating the hot mess that is tablefield patching, and accepting of the risks involved there.

A faster/simpler option would be to write a custom constraint (form validation) that would present an error if the first row is blank. This would not require collab with the community, as it doesn't change the behavior of the module at all. It does, however, come at a cost of some tech debt as we would need to maintain this constraint over time. This approach would be my recommendation as it can be done without impacting the future state of the module, and it can be easily removed if/when the tablefield module moves header assignments to separate fields. I would say a roughly 3 point ticket for doing this would be adequate.

**Comment by wesrowe** [comment]
@dsasser, thanks for the thoughtful response. The UI constraint sounds like a more viable option. However, the tech debt is a concern.

@laflannery, I would like to wait to do anything in the code until we hear back from the audit how frequently a blank top row occurs in the wild. 

The no-brainer first move is to change the help text in the CMS UI so that we don't give people the idea. I'll write a ticket.

**Comment by wesrowe** [comment]
Hi @laflannery, just checking in to see how the content audit is progressing. We changed the helper text in the CMS to say that headers are required, but we didn't implement the requirement in code pending how big a problem it is in the wild.

**Comment by laflannery** [comment]
@wesrowe the updated CMS table audit, which is what we needed to do a content audit was just finished so we will be able to efficiently start a full content audit now.  I need to set a meeting with Randi to review the table audit.

**Comment by jilladams** [comment]
@laflannery @wesrowe I missed this one in my round up on https://github.com/department-of-veterans-affairs/va.gov-cms/issues/10800#issuecomment-1318650289. Also closeable?

**Comment by laflannery** [comment]
@jilladams Yes, I think it can be closed.  Blank headings seem to occur infrequently and in the majority of cases it was when tables were being used as layout tables (which is not correct). I think the updated helper text you already did is a good solution for this issue.

**Comment by jilladams** [comment]
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11025#issuecomment-1322098269
